### ISSUE TITLE
Allow the layout to be appended to any element other than the body

### ISSUE THREAD

**Issue by vegegoku** [DominoKit/domino-ui#88]

**Comment by vegegoku** [comment]
@ainslec doing a quick research i found it is better to leave the layout as is - maybe we name it `AppLayout` - and introduce a new set of layouts based on css grids
https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Grid_Layout

**Comment by ainslec** [comment]
Keeping them separate makes sense. Working around this is a bit tough at the moment. I look forward to seeing the css grid approach, when you have time :-)

**Comment by vegegoku** [comment]
@ainslec i am doing small experiments when i find some time, but i doubt i will have something useful during the vacation so it might take some time.

**Comment by ainslec** [comment]
Enjoying your holiday is your first priority. Stop checking your mail !

**Comment by vegegoku** [comment]
To provide a better layout option we rolled a huge update in domino-ui :

- Add support for 12,16,18,24,32 grid systems [demo-page](https://dominokit.github.io/domino-ui-demo/index.html?theme=indigo#grids)
- Add support for column offset, this allow the column to start from  a specific cell in a row `Column.span4().offset2()` for example will start the column from the 3rd cell and will span over 4 cells.
- Add GridLayout that has 5 section , content, header, footer, right and left with custom span based on 12 column/rows grid system [demo-page](https://dominokit.github.io/domino-ui-demo/index.html?theme=indigo#grid-layout)
- All domino-ui components now extends from a base class with most basic functionality, including for example : `dominoComponent.onAttach(...)`, `dominoComponent.setAttribute(...)`, `dominoElement.appendChild(...)`,`dominoElement.addClickListener(...)`,`dominoElement.style()..` ...etc, more functions will be added over time to abstract the use of domino-ui components and normal elemental2 elements.
- We used to have methods in domino-ui component like `appendContent`, `addItem`,`appendItem`,`addXXX` ..etc which leads to inconsistency in the API, and gives the user hard to figure what he should call to append another element, and if he should call asElement or not, which in some cases also lead to have unneeded .asElement() calls and clutters the code. Now we **DEPRECATED** many of those methods and replaced then with overloaded versions of `appendChild`, this will hugely reduce the need to call `.asElement()` and will make appending consistence`
